### DISCUSSION TITLE
1.5.0 / 1.5.0-beta.1

### DISCUSSION THREAD

**Discussion by utpilla** [open-telemetry/opentelemetry-dotnet#4546] [Announcements]
**OpenTelemetry**

* Fixed a bug introduced by
  [#4508](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4508) in
  1.5.0-rc.1 which caused the "Build" extension to return `null` when performing
  chained/fluent calls.
  ([#4529](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4529))

* Marked `Exemplars` and related APIs `internal` as the spec for `Exemplars` is
  not stable yet. This would be added back in the `1.6.*` prerelease versions
  right after `1.5.0` stable version is released.
  ([#4533](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4533))

**OpenTelemetry.Api.ProviderBuilderExtensions**

* Added an `IServiceCollection.ConfigureOpenTelemetryMeterProvider` overload
  which may be used to configure `MeterProviderBuilder`s while the
  `IServiceCollection` is modifiable (before the `IServiceProvider` has been
  created).
  ([#4517](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4517))

**OpenTelemetry.Exporter.Console**

* Remove support for exporting `Exemplars`. This would be added back in the
  `1.6.*` prerelease versions right after `1.5.0` stable version is released.
  ([#4533](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4533))

**OpenTelemetry.Exporter.OpenTelemetryProtocol**

* Remove support for exporting `Exemplars`. This would be added back in the
  `1.6.*` prerelease versions right after `1.5.0` stable version is released.
  ([#4533](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4533))

**OpenTelemetry.Instrumentation.AspNetCore**

* Fix issue where baggage gets cleared when the ASP.NET Core Activity
   is stopped. The instrumentation no longer clears baggage. One problem
   this caused was that it prevented Activity processors from accessing baggage
   during their `OnEnd` call.
([#4274](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4274))

* Added direct reference to `System.Text.Encodings.Web` with minimum version of
`4.7.2` due to [CVE-2021-26701](https://github.com/dotnet/runtime/issues/49377).
This impacts target frameworks `netstandard2.0` and `netstandard2.1` which has a
reference to `Microsoft.AspNetCore.Http.Abstractions` that depends on
`System.Text.Encodings.Web` >= 4.5.0.
([#4399](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4399))

* Improve perf by avoiding boxing of common status codes values.
  ([#4360](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4360),
  [#4363](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4363))

**OpenTelemetry.Instrumentation.Http**

* Fixed an issue of missing `http.client.duration` metric data in case of
  network failures (when response is not available).
  ([#4098](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4098))

* Improve perf by avoiding boxing of common status codes values.
  ([#4361](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4361),
  [#4363](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4363))

<hr /><em>This discussion was created from the release <a href='https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.5.0'>1.5.0 / 1.5.0-beta.1</a>.</em>

**Comment by tomotvos** [comment]
Looking over the changes, it seems like 1.5 is primarily a bug-fix version over 1.4. Do I have that right? Is there anything that is new?

**Reply by utpilla** [reply]
@tomotvos The major feature that got added in 1.5.0 is the support for Exponential Histograms.

The changes shown here are added on top of what was already added to the previous 1.5.0-rc.1 release. To look for all the changes that were added in 1.5.0, you would have to look at the changes made in all the previous 1.5 prerelease versions as well. (1.5.0-rc.1, 1.5.0-alpha.2, 1.5.0-alpha.1).

**Reply by tomotvos** [reply]
Ok thanks. Is there another README somewhere that mentions this, or is the expectation that we UNION all the changes ourselves? Asking for a friend...
